### ISSUE TITLE
Byte compile produces warnings for unknown functions

### ISSUE THREAD

**Issue by Forge36** [andras-simonyi/citeproc-el#165]
```
 ■  Warning (comp): citeproc-s.el:240:21: Warning: the function ‘citeproc-s-replace’ is not known to be defined.
 ■  Warning (comp): citeproc-lib.el:159:33: Warning: the function ‘citeproc-rt-render-affixes’ is not known to be defined.
 ■  Warning (comp): citeproc-lib.el:154:55: Warning: the function ‘citeproc-context-vars’ is not known to be defined.
 ■  Warning (comp): citeproc-rt.el:538:25: Warning: the function ‘citeproc-itemdata-rawbibitem’ is not known to be defined.
 ■  Warning (comp): citeproc-prange.el:40:44: Warning: the function ‘citeproc-s-fill-copy’ is not known to be defined.
 ■  Warning (comp): citeproc-date.el:271:13: Warning: the function ‘citeproc-number--format-as-ordinal’ is not known to be defined.
 ■  Warning (comp): citeproc-formatters.el:384:48: Warning: the function ‘citeproc-rt-to-plain’ is not known to be defined.
 ■  Warning (comp): citeproc-formatters.el:164:21: Warning: the function ‘citeproc-s-replace-all-seq’ is not known to be defined.
 ■  Warning (comp): citeproc-sort.el:90:19: Warning: the function ‘citeproc-number-var-value’ is not known to be defined.
```

**Comment by Forge36** [comment]
To my knowledge the [native compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Native-Compilation.html) is automatic (provide emacs is built with support. I'm not able to trigger this error with `byte-recompile-file`

System details:
> I have an init file
> (emacs-version)
> GNU Emacs 29.3 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0) of 2024-05-10
>
> OS Windows 10 in WSL (Ubuntu 22.04.4 LTS)

Some testing to reproduce:
I modify `citeproc-sort.el` restart emacs, and insert a citation in org triggers the async-compile.
I can fix this error by adding `(require 'citeproc-number)`

I'm not having luck using `emacs -q` to produce a minimal test case unfortunately (I'm struggling to use emacs in that scenario unfortunately).

I'm going to poke at an MR to address these if I can determine how (I'm new to upstreaming fixes apologies if I'm not following proper etiquette).

**Comment by Forge36** [comment]
This appears fixed with the latest #169!

Closing
